pkg/istio/virtualservice_test.go

```diff
@@ -132,215 +265,350 @@ var _ = Describe("Operations for istio VirtualService kind", func() {
+					ref.AddTargetResource(targetV1)
+					ref.AddTargetResource(model.NewLocatedResource("Service", "details", map[string]string{}))
+
+					err := VirtualServiceMutator(ctx, &ref)
 					Expect(err).ToNot(HaveOccurred())
 
-					Expect(GetMutatedRoute(mutatedVirtualService, targetV1Host, targetV1Subset).Redirect).To(BeNil())
+					virtualService := get.VirtualService("test", "details")
+					Expect(GetMutatedRoute(virtualService, targetV1Host, targetV1Subset).Redirect).To(BeNil())
 				})
+
 				It("include destinations with no subset", func() {
-					mutatedVirtualService, err = mutateVirtualService(ctx, targetV6Host, targetV6.Labels["version"], targetV6Subset, virtualService)
-					Expect(err).ToNot(HaveOccurred())
+					ref.AddTargetResource(targetV6)
+					ref.Name = "x-v5"
+					ref.AddTargetResource(model.NewLocatedResource("Service", "x", map[string]string{}))
 
-					Expect(GetMutatedRoute(mutatedVirtualService, targetV6Host, targetV6Subset).Redirect).To(BeNil())
-				})
+					err := VirtualServiceMutator(ctx, &ref)
+					Expect(err).ToNot(HaveOccurred())
 
-				It("route missing", func() {
-					_, err = mutateVirtualService(ctx, model.HostName{Name: "miss-v5"}, "v5", "v5-vs-test", virtualService)
-					Expect(err).To(HaveOccurred())
-					Expect(err.Error()).To(ContainSubstring("route not found"))
-				})
-				It("route missing version", func() {
-					_, err = mutateVirtualService(ctx, model.HostName{Name: "details-v10"}, "v10", "v10-vs-test", virtualService)
-					Expect(err).To(HaveOccurred())
-					Expect(err.Error()).To(ContainSubstring("route not found"))
+					virtualService := get.VirtualService("test", "details")
+					Expect(GetMutatedRoute(virtualService, targetV6Host, targetV6Subset).Redirect).To(BeNil())
 				})
 			})
 		})
 
 		Context("revertors", func() {
 
-			JustBeforeEach(func() {
-				err = k8yaml.Unmarshal([]byte(yaml), &virtualService)
-				Expect(err).ToNot(HaveOccurred())
+			BeforeEach(func() {
+				objects = []runtime.Object{
+					&istionetwork.VirtualService{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "details",
+							Namespace: "test",
+						},
+						Spec: istionetworkv1alpha3.VirtualService{
+							Hosts: []string{"details"},
+							Http: []*istionetworkv1alpha3.HTTPRoute{
+								&istionetworkv1alpha3.HTTPRoute{
+									Route: []*istionetworkv1alpha3.HTTPRouteDestination{
+										&istionetworkv1alpha3.HTTPRouteDestination{
+											Destination: &istionetworkv1alpha3.Destination{
+												Host:   "details",
+												Subset: "v1-vs-test",
+											},
+										},
+									},
+									Match: []*istionetworkv1alpha3.HTTPMatchRequest{
+										&istionetworkv1alpha3.HTTPMatchRequest{
+											Headers: map[string]*istionetworkv1alpha3.StringMatch{
+												"x-test-suite": &istionetworkv1alpha3.StringMatch{MatchType: &istionetworkv1alpha3.StringMatch_Exact{Exact: "feature-x"}},
```

Comment:
I recall you were showing me some "builders" for those objects? Or was that early stage of `Helpers`?

Reply:
We have the scenario generators but that's different. I guess we could expand on them to use them here too.

The `Helpers` are just "Type-safe" wrappers for getting RuntimeObjects from the Client.